url0980c8326f_TobiasScharnowski_Chess_tgz-pJ8-chess_StartJ8/src/chess/move/impl/PGNParser.java

```diff
@@ -0,0 +1,185 @@
+package chess.move.impl;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import chess.ChessCoord;
+import chess.ChessRules;
+import chess.PieceType;
+import chess.Position;
+import chess.impl.Coords;
+import chess.move.Move;
+import chess.move.MoveParser;
+import chess.piece.ChessColor;
+
+/**
+ * {@link chess.move.MoveParser MoveParser} implementation for Portable Game
+ * Notation (PGN).
+ */
+public class PGNParser implements MoveParser {
+    private static final char INDICATOR_PAWN_PROMOTION = '=';
+    private static final char INDICATOR_CHECK = '+';
+    private static final char INDICATOR_TAKING_MOVE = 'x';
+    private static final String CASTLING = "O-O";
+    private static final String CASTLING_LONG = "O-O-O";
+    private static final int COORD_NOTATION_LENGTH = 2;
+
+    @Override
+    public Move parseMove(Position pos, String notation) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 18. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.